pyccl/boltzmann.py

```diff
@@ -174,6 +181,166 @@ def get_camb_pk_lin(cosmo):
     return pk_lin
 
 
+def get_isitgr_pk_lin(cosmo):
+    """Run ISiTGR-CAMB and return the linear power spectrum.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): Cosmological
+            parameters. The cosmological parameters with
+            which to run ISiTGR-CAMB.
+
+    Returns:
+        :class:`~pyccl.pk2d.Pk2D`: Power spectrum \
+            object. The linear power spectrum.
+    """
+
+    assert HAVE_ISITGR, (
+        "You must have the `isitgr` python package "
+        "installed to run CCL with ISiTGR-CAMB!")
```

Comment:
Don't do this.  Bare `assert` statements should only be used for things that you as the programmer thinks should always be true, and if they ever hit it means you screwed up.  Not that your user did something wrong.  (Like try to use boltzmann_isitgr without having isitgr installed.)

Rather, I'd recommend moving the try/except block at the top down to here and reraising if necessary with extra information:
```
try:
    import isitgr
    import isitgr.model
except ImportError as e:
    e.args = (
        "You must have the `isitgr` python package "
        "installed to run CCL with ISiTGR-CAMB!",
        *e.args)
    raise
```

Reply:
I followed what was there already for camb for a while. I replicated the same things for isitgr. So this may need some attention but later for all of them.

Reply:
I will send more comments tomorrow.

Reply:
IMO, it's bad design for all of them then.  Better to do it right on this one and put in a `# TODO clean up other assert anti-patterns in this file`

Reply:
Functionally these different variants are the same. The asserts in the code give the proper debugging information and are not bare with no message at all. The only difference is which error is raised. 

Feel free to change them if you’d like. 

The real anti pattern here is an assert with no error message text at all.

Reply:
No, they aren't functionally the same.  The problem with `assert` statements is that they are disableable.  So they should really only be used for things that you think should absolutely never ever happen.  Things like loop invariants.  They are helpful for laying out what you think the logic of an algorithm is.  If the assert fails, it means you screwed up the logic somehow, and you need to fix your code.

This is not that.  It is user error.  The appropriate kind of exception to raise really is `ImportError`, not `AssertionError`.  And you don't want the user to be able to turn it off with a command-line option.  You always want this checked.

More discussion about why I (and others) consider this an anti-pattern:
https://github.com/emre/notes/blob/master/python/when-to-use-assert.md
https://dev.to/yloganathan/do-not-abuse-the-assert-1cfm
https://stackoverflow.com/a/945135/1332281

Reply:
I changed this for isitgr as suggested and I put a comment there. 

Travis is now passing. Could you tell me what the message below is about as it is failing:

coverage/coveralls — Coverage decreased (-0.09%) to 96.294%

see further below as well.

Reply:
It means you need to test your new code.  You don't have any unit tests of the code you added! (Benchmarks don't count -- you need code in the tests directory that verify that the code you added does what you think it does.

Reply:
Hum. I have a jupyter notebook (a PR in CCLX as there where we put examples now) that uses the new code and produces the power spectrum and the Cls and correlations as it should. That is how I tested that the code does what it is supposed to do.  Will that count then? and if so how to integrated that here.

Reply:
Let me know and we will be able to wrap this up.

Reply:
No, that doesn't count.  You need to write unit tests.  Look at the other things in the tests directory.  Also, maybe watch this: 
https://www.youtube.com/watch?v=u3x5OEXgtnU&list=PLnq2BjRWgmjtCZwGyJGyFb_ig9Js1oAkg&index=2

Reply:
I see. I just located the tests under pyccl. Interesting. I will watch the video tomorrow (1 hour 24 min!). 
I am trying to understand the big picture for now. Can you you clarify: 
1) is there some other engine (beside travis) here that is checking and finding that there are no tests. Can you clarify what is it or expalin this part.
2) How to interprete the message received? which files does it complain about. For example, I don't recall changing pyccl/pyutils.py.